src/js/node/http.ts

```diff
@@ -953,11 +957,43 @@ class OutgoingMessage extends Writable {
   }
 }
 
+function emitCloseNT(self) {
+  if (!self._closed) {
+    self.destroyed = true;
+    self._closed = true;
+    self.emit("close");
+  }
+}
+
+function onServerResponseClose() {
+  // EventEmitter.emit makes a copy of the 'close' listeners array before
+  // calling the listeners. detachSocket() unregisters onServerResponseClose
+  // but if detachSocket() is called, directly or indirectly, by a 'close'
+  // listener, onServerResponseClose is still in that copy of the listeners
+  // array. That is, in the example below, b still gets called even though
+  // it's been removed by a:
+  //
+  //   const EventEmitter = require('events');
+  //   const obj = new EventEmitter();
+  //   obj.on('event', a);
+  //   obj.on('event', b);
+  //   function a() { obj.removeListener('event', b) }
+  //   function b() { throw "BAM!" }
+  //   obj.emit('event');  // throws
+  //
+  // Ergo, we need to deal with stale 'close' events and handle the case
+  // where the ServerResponse object has already been deconstructed.
+  // Fortunately, that requires only a single if check. :-)
+  if (this._httpMessage) {
+    emitCloseNT(this._httpMessage);
+  }
+}
+
 let OriginalWriteHeadFn, OriginalImplicitHeadFn;
-class ServerResponse extends Writable {
+class ServerResponse extends OutgoingMessage {
   declare _writableState: any;
 
-  constructor({ req, reply }) {
+  constructor(req, reply) {
```

Comment:
My only worry is that in Node, this `reply` is an object of `options` (though i couldnt find any docs on it). I think this should be fine though.

Reply:
https://github.com/nodejs/node/blob/v21.3.0/lib/_http_server.js#L195